tests/fixtures/policies/dummy_policy.py

```diff
@@ -88,9 +87,29 @@ def vectorized(self):
         """
         return True
 
+    @property
+    def observation_space(self):
```

Comment:
Aren't these the default definitions? Why are they repeated here?

Reply:
One major issue I encountered is that, we can't make `garage.np.policy` an abstract class as `garage.tf.policy` has to inherit from another class. We can only make `garage.np.policy` an interface.

Reply:
Do you think there is a workaround/better way to handle this?

Reply:
I'm not quite sure I understand the issue, to be honest. `garage.np.Policy` *does* inherit from `abc.ABC`, so I thought that made it an abstract class. Also, `DummyPolicy` here has nothing to do with `garage.tf.Policy`, so I'm doubly confused. Does anything break if you delete this method definition on `DummyPolicy`?

Reply:
oh, it's just `garage.np.Policy` cannot have a constructor in `TensorFlow` branch, so we can't have default property.

Reply:
Yeah, some tests need `DummyPolicy.observation_space`.

Reply:
Oh, I see what you're getting at. If we're going to have a public `env_spec` property, couldn't we just have default definitions for these properties that use the `.env_spec` property?

Reply:
are you suggesting:

```py
# garage.tf.Policy
import numpy as np

from garage import Policy as IPolicy


class Policy(IPolicy):

    def __init__(self, env_spec, name):
        self._name = name
        self._env_spec = env_spec

    @property
    def observation_space(self):
        return self._env_spec.observation_space

    @property
    def action_space(self):
        return self._env_spec.action_space

    def get_action(self, observation):
        return self.get_actions(np.expand_dims(observation))[0]

    def get_reset_state(self):
        return self.get_reset_states(1)[0]
```

Reply:
So I think that `get_reset_state` and friends are way outside the scope of this PR. I'm just trying to give some minor suggestions here.

I don't actually think we need the `env_spec`, `observation_space`, or `action_space` properties, to be honest. The entire torch branch currently doesn't use them (and there doesn't really even seem to be anywhere they would be useful to use). Mostly this is just because I strongly prefer to not have unused (or mostly unused) methods in regularly implemented APIs.

However, in the interest of making a small, simple change with this PR, I think the following makes sense, at least.

```python

# garage.np.Policy

class Policy(abc.ABC):

    @abc.abstractproperty
    def env_spec(self):
        pass

    @property
    def observation_space(self):
        return self.env_spec.observation_space

    @property
    def action_space(self):
        return self.env_spec.action_space

# tests.fixtures.policies.DummyPolicy

from garage.np.policies import Policy as BasePolicy

class DummyPolicy(BasePolicy):

    def __init__(self, env_spec):
        self._env_spec = env_spec

    @property
    def env_spec(self):
        return self._env_spec
```

Reply:
Honestly, I was just trying to avoid some duplicated code with this comment. I don't really get what's hard here.

Reply:
Currently in tf tree, observation_space and action_space are often used (e.g. to flatten / unflatten observations) so i think maybe it's good time to figure out if it should be included in the interface later when we really refactor the policy interface.

I think you proposed a good way to eliminate duplicated code specifically for the existing policy interface, although we will change it later anyway. For this PR, I think the objective is to unify policy interface across the codebase and remove duplicated code.

Reply:
@krzentner sorry i didn't mean we would implement the interface in this PR, just that quibbling over the details for a long time doesn't matter much if we are going to rip out much of this in the next ~2 weeks